pkg/util/kubernetes/util.go

```diff
@@ -228,3 +232,37 @@ func ResolveValueSource(ctx context.Context, client k8sclient.Reader, namespace
 
 	return "", nil
 }
+
+// GetTolerations build an array of Tolerations from an array of string
+func GetTolerations(taints []string) ([]corev1.Toleration, error) {
```

Comment:
May I suggest to move it into a dedicated file? The `util.go` has grown too big IMHO.

Reply:
I think it is appropriate keeping as it is. I'll explain why. You can see that such utility unit is bundling different methods treating `corev1` structs. I tried to be as much cohesive as possible. If I use another utility unit I feel I break that cohesion just for the purpose of reducing the size. Finally I preferred to have better cohesion versus the lenght of the `util`. Did I convince you?

Reply:
Yes and no 😉.

I understand it already contains client methods for other resources, and few other miscellaneous
methods like: `ToJSON`, `ToYAML` `ResolveValueSource` `GetSecretRefValue`. I'd argue that the main difference is that `GetTolerations(taints []string)` is a factory method, while the other, resources related methods, are actually client getters. So yes, because it already breaks the single responsibility principle and it's not a big deal :) and no because this is not a good reason to introduce another kind of method to it 😄.

Reply:
Haha, yeah, those are valid points too. With 8118aae I tried to refactor in order to have a tidier organization. I am not entirely sure about the names though. What do you think @astefanutti?

Reply:
What I could propose is:
- Move all the _client_ methods, that use `context context.Context, client k8sclient.Reader`, in a `client.go` or `reader.go` file
- Move the `GetTolerations` into either a `factory.go` file, if we plan to add more factory methods or group the existing ones, or a `toleration.go` file, if we want to be more granular.

I understand this is opinion when it comes to naming and grouping, so I'll trust and respect your choice :)

Reply:
Makes sense, just fixed.